src/components/SchemaRow/TopLevelSchemaRow.tsx

```diff
@@ -1,85 +1,107 @@
+import { useOnScreen } from '../../hooks/useOnScreen';
 import { calculateChildrenToShow, isComplexArray } from '../../tree';
+import { showPathCrumbsAtom } from '../PathCrumbs';
 import { ChildStack } from '../shared/ChildStack';
 import { SchemaRow, SchemaRowProps } from './SchemaRow';
 import { useChoices } from './useChoices';
 
-export const TopLevelSchemaRow: React.FC<SchemaRowProps> = ({ schemaNode, nestingLevel }) => {
+export const TopLevelSchemaRow = ({ schemaNode }: Pick<SchemaRowProps, 'schemaNode'>) => {
   const { selectedChoice, setSelectedChoice, choices } = useChoices(schemaNode);
   const childNodes = React.useMemo(() => calculateChildrenToShow(selectedChoice.type), [selectedChoice.type]);
+  const nestingLevel = 0;
 
   // regular objects are flattened at the top level
   if (isRegularNode(schemaNode) && isPureObjectNode(schemaNode)) {
     return (
-      <DecreaseIndentation>
+      <>
+        <ScrollCheck />
         <ChildStack childNodes={childNodes} currentNestingLevel={nestingLevel} />
-      </DecreaseIndentation>
+      </>
     );
   }
 
   if (isRegularNode(schemaNode) && choices.length > 1) {
     const combiner = isRegularNode(schemaNode) && schemaNode.combiners?.length ? schemaNode.combiners[0] : null;
 
     return (
-      <DecreaseIndentation>
-        <Box pos="relative">
+      <>
+        <ScrollCheck />
+
+        <HStack spacing={3} pb={4}>
           <Menu
             aria-label="Pick a type"
             closeOnPress
-            matchTriggerWidth
+            placement="bottom left"
             items={choices.map((choice, index) => ({
               id: index,
               title: choice.title,
               onPress: () => setSelectedChoice(choice),
             }))}
             renderTrigger={props => (
               <Pressable {...props}>
-                <Flex>
-                  <Flex mr={2} fontFamily="mono" fontWeight="semibold" fontSize="base" cursor="pointer" py={2}>
-                    {selectedChoice.title}
-                    <Box ml={1}>
-                      <Icon icon={faCaretDown} />
-                    </Box>
-                  </Flex>
-                  {combiner !== null ? (
-                    <Flex alignItems="center" color="muted">
-                      {combiner}
-                    </Flex>
-                  ) : null}
+                <Flex fontFamily="mono" fontWeight="semibold" cursor="pointer" fontSize="base">
+                  {selectedChoice.title}
+                  <Box ml={1}>
+                    <Icon icon={faCaretDown} />
+                  </Box>
                 </Flex>
               </Pressable>
             )}
           />
 
-          {childNodes.length > 0 ? <ChildStack childNodes={childNodes} currentNestingLevel={nestingLevel} /> : null}
-        </Box>
-      </DecreaseIndentation>
+          {combiner !== null ? (
+            <Flex alignItems="center" color="muted" fontSize="base">
+              {`(${COMBINER_NAME_MAP[combiner]})`}
```

Comment:
At first I thought this wasn't working for some schemas:

Working:
<img width="432" alt="Screen Shot 2021-11-30 at 4 50 20 PM" src="https://user-images.githubusercontent.com/3588798/144141118-6a09f46a-eb4d-486f-8670-22bb67cfda6a.png">

Not working:
<img width="410" alt="Screen Shot 2021-11-30 at 4 50 30 PM" src="https://user-images.githubusercontent.com/3588798/144141575-05d61d22-6e7d-42ce-83ce-9da01859bc67.png">

But then I realized that although the schema has a `oneOf` key, it's nested in the `items` object. 🤦‍♂️ https://github.com/stoplightio/json-schema-viewer/blob/3ceb51e692f709784774964cab6d119ebd38a235/src/__fixtures__/combiners/oneof-within-array-item.json#L3-L5

I'm not sure what should happen in this case. Is the schema wrong/invalid? Or do we just not show the combiner name for this schema and call it good?

Reply:
Hmm I think this has been an issue on main branch (not introduced by this PR) - can fix it in this PR or we can follow-up in a separate PR. What do you think?

Reply:
I'm fine with a followup PR!